src/coreclr/jit/lower.cpp

```diff
@@ -550,13 +550,9 @@ GenTree* Lowering::LowerNode(GenTree* node)
                 return nextNode;
             }
 
-            nextNode = LowerCast(node);
-            if (nextNode != nullptr)
-            {
-                return nextNode;
-            }
```

Comment:
This is reverting a change from #97529. The new implementation always preserves the original cast node and does all IR manipulation ahead of it.